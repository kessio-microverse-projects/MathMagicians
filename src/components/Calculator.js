import React, { useState } from 'react';
import DisplayValues from './DisplayValues';
import calculate from '../logic/Calculate'; /* eslint-disable-line */
import '../styles/calculator.css';

const Calculator = () => {
  const [obj, setCalculations] = useState({ total: null, next: null, operation: null });

  const clickOperations = (event) => {
    setCalculations(calculate(obj, event.target.value));
  };

  return (
    <div>
      <div className="sectiontitle">Let us do some Maths</div>
      <div className="appCalculator">
        <div className="grid-container">
          <div className="screen-input">
            <span className="text-display">
              <DisplayValues total={obj.total} operation={obj.operation} next={obj.next} />
            </span>
          </div>
        </div>
        <div className="grid-container">
          <input type="button" data-testid="ac" onClick={clickOperations} value="AC" className="same-size-btn" />
          <input type="button" data-testid="plus" onClick={clickOperations} value="+/-" className="same-size-btn" />
          <input type="button" data-testid="percent" onClick={clickOperations} value="%" className="same-size-btn" />
          <input type="button" data-testid="divide" onClick={clickOperations} value="รท" className="same-size-btn orange" />
        </div>
        <div className="grid-container">
          <input type="button" data-testid="seven" onClick={clickOperations} value="7" className="same-size-btn" />
          <input type="button" data-testid="eight" onClick={clickOperations} value="8" className="same-size-btn" />
          <input type="button" data-testid="nine" onClick={clickOperations} value="9" className="same-size-btn" />
          <input type="button" data-testid="multiply" onClick={clickOperations} className="same-size-btn orange" value="x" />
        </div>
        <div className="grid-container">
          <input type="button" data-testid="four" onClick={clickOperations} value="4" className="same-size-btn" />
          <input type="button" data-testid="five" onClick={clickOperations} value="5" className="same-size-btn" />
          <input type="button" data-testid="six" onClick={clickOperations} value="6" className="same-size-btn" />
          <input type="button" data-testid="minus" onClick={clickOperations} value="-" className="same-size-btn orange" />
        </div>
        <div className="grid-container">
          <input type="button" data-testid="one" onClick={clickOperations} value="1" className="same-size-btn" />
          <input type="button" data-testid="two" onClick={clickOperations} value="2" className="same-size-btn" />
          <input type="button" data-testid="three" onClick={clickOperations} value="3" className="same-size-btn" />
          <input type="button" data-testid="plus" onClick={clickOperations} value="+" className="same-size-btn orange" />
        </div>
        <div className="grid-container">
          <input type="button" data-testid="zero" onClick={clickOperations} className="zero" value="0" />
          <input type="button" data-testid="decimal" onClick={clickOperations} value="." className="same-size-btn" />
          <input type="button" data-testid="equals" onClick={clickOperations} value="=" className="same-size-btn orange" />
        </div>
      </div>
    </div>
  );
};

export default Calculator;
