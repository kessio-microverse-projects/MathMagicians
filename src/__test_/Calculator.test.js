import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../components/Calculator';

describe('snapshot test for Calculator Component', () => {
it('Matches the snapshot', () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot('Calculator');
})
});