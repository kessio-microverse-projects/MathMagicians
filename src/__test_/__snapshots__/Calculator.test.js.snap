// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test for Calculator Component Matches the snapshot 1`] = `
<div>
  <div
    className="sectiontitle"
  >
    Let us do some Maths
  </div>
  <div
    className="appCalculator"
  >
    <div
      className="grid-container"
    >
      <div
        className="screen-input"
      >
        <span
          className="text-display"
        >
          <div />
        </span>
      </div>
    </div>
    <div
      className="grid-container"
    >
      <input
        className="same-size-btn"
        data-testid="ac"
        onClick={[Function]}
        type="button"
        value="AC"
      />
      <input
        className="same-size-btn"
        data-testid="plus"
        onClick={[Function]}
        type="button"
        value="+/-"
      />
      <input
        className="same-size-btn"
        data-testid="percent"
        onClick={[Function]}
        type="button"
        value="%"
      />
      <input
        className="same-size-btn orange"
        data-testid="divide"
        onClick={[Function]}
        type="button"
        value="รท"
      />
    </div>
    <div
      className="grid-container"
    >
      <input
        className="same-size-btn"
        data-testid="seven"
        onClick={[Function]}
        type="button"
        value="7"
      />
      <input
        className="same-size-btn"
        data-testid="eight"
        onClick={[Function]}
        type="button"
        value="8"
      />
      <input
        className="same-size-btn"
        data-testid="nine"
        onClick={[Function]}
        type="button"
        value="9"
      />
      <input
        className="same-size-btn orange"
        data-testid="multiply"
        onClick={[Function]}
        type="button"
        value="x"
      />
    </div>
    <div
      className="grid-container"
    >
      <input
        className="same-size-btn"
        data-testid="four"
        onClick={[Function]}
        type="button"
        value="4"
      />
      <input
        className="same-size-btn"
        data-testid="five"
        onClick={[Function]}
        type="button"
        value="5"
      />
      <input
        className="same-size-btn"
        data-testid="six"
        onClick={[Function]}
        type="button"
        value="6"
      />
      <input
        className="same-size-btn orange"
        data-testid="minus"
        onClick={[Function]}
        type="button"
        value="-"
      />
    </div>
    <div
      className="grid-container"
    >
      <input
        className="same-size-btn"
        data-testid="one"
        onClick={[Function]}
        type="button"
        value="1"
      />
      <input
        className="same-size-btn"
        data-testid="two"
        onClick={[Function]}
        type="button"
        value="2"
      />
      <input
        className="same-size-btn"
        data-testid="three"
        onClick={[Function]}
        type="button"
        value="3"
      />
      <input
        className="same-size-btn orange"
        data-testid="plus"
        onClick={[Function]}
        type="button"
        value="+"
      />
    </div>
    <div
      className="grid-container"
    >
      <input
        className="zero"
        data-testid="zero"
        onClick={[Function]}
        type="button"
        value="0"
      />
      <input
        className="same-size-btn"
        data-testid="decimal"
        onClick={[Function]}
        type="button"
        value="."
      />
      <input
        className="same-size-btn orange"
        data-testid="equals"
        onClick={[Function]}
        type="button"
        value="="
      />
    </div>
  </div>
</div>
`;
